pkg/api/types.go

```diff
@@ -2141,6 +2141,22 @@ const (
 
 	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
 	SSHAuthPrivateKey = "ssh-privatekey"
+
+	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
+	//
+	// Required fields:
+	// - Secret.Data["key"] - TLS private key.
+	//   Secret.Data["cert"] - TLS certificate.
+	//   Secret.Data["chain"] - optional certificate chain.
+	// TODO: Consider supporting different formats, specifying CA/destinationCA.
+	SecretTypeTLS SecretType = "kubernetes.io/tls"
+
+	// TLSCertKey is the key for tls certificates in a TLS secert.
+	TLSCertKey = "cert"
+	// TLSPrivateKeyKey is the key for the private key field in a TLS secret.
+	TLSPrivateKeyKey = "key"
+	// TLSChainKey is the key for the certificate chain in a TLS secret.
+	TLSChain = "chain"
```

Comment:
It can be used as a chain for this server's cert (in which case they could shove it all in the cert), for client authentication, or a re-encrypt handshake. Also having it separate makes defaulting it easier. 

On sites like godaddy you just pick your webserver version and it gives you what it thinks is right. So if you pick apache 2.2, you'll get a chain, cert and key. Less cognitive friction by having an optional third field? 

But if it's ambiguous I can leave it out of the first cut and add it later if needed.